core/vm_env.go

```diff
@@ -41,29 +41,38 @@ func GetHashFn(ref common.Hash, chain *BlockChain) func(n uint64) common.Hash {
 }
 
 type VMEnv struct {
-	state  *state.StateDB
+	// State to use for executing
+	state *state.StateDB
+	// Header information
 	header *types.Header
-	msg    Message
-	depth  int
-	chain  *BlockChain
-	typ    vm.Type
-
-	getHashFn func(uint64) common.Hash
-	// structured logging
+	// Message appliod
```

Comment:
No, just that generally writing comments after fields makes the code more condensed and easier to skim through. It also aligns the comments nicely.

Usually with interfaces the method name is already long enough so adding comment after it just makes it unreadable and also method comments tend to be much longer, usually multiple lines whereas field comments are only a few words.